pkg/policy/rule.go

```diff
@@ -158,9 +191,20 @@ func (r *rule) resolveL3Policy(ctx *SearchContext, state *traceState, result *L3
 }
 
 func (r *rule) canReach(ctx *SearchContext, state *traceState) api.Decision {
+	entitiesDecision := r.canReachEntities(ctx, state)
+
 	if !r.EndpointSelector.Matches(ctx.To) {
-		ctx.PolicyTraceVerbose("  Rule %d %s: no match for %+v\n", state.ruleID, r, ctx.To)
-		return api.Undecided
+		if entitiesDecision == api.Undecided {
+			ctx.PolicyTraceVerbose("  Rule %d %s: no match for %+v\n", state.ruleID, r, ctx.To)
+		} else {
+			state.selectedRules++
+			ctx.PolicyTrace("* Rule %d %s: match\n", state.ruleID, r)
+		}
+		return entitiesDecision
+	}
+
+	if entitiesDecision == api.Denied {
```

Comment:
This is dead code, `canReachEntities` does not and should never return `api.Denied`

Reply:
Indeed, sorry about that.